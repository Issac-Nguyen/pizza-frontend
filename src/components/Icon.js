import React from 'react';
import PropTypes from 'prop-types';

const Icon = (props) => {
    const {name, ...other} = props;
    return (
        <i data-testid="icon" className={`las ${name}`} {...other}></i>
    )
}

Icon.propTypes = {
    name: PropTypes.string.isRequired,
}

export default Icon;