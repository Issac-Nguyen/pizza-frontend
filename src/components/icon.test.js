import React from 'react'
import {render, cleanup} from '@testing-library/react'
import Icon from './Icon'

describe('Icon', () => {
    afterEach(cleanup);
 
    it('should take a snapshot', () => {
       const { asFragment } = render(<Icon name="test"/>)
       
       expect(asFragment(<Icon name="test"/>)).toMatchSnapshot()
    })

    it('should have icon', () => {
        const { container  } = render(<Icon name="test"/>)
        
        expect(container.querySelector('i')).toBeInTheDocument();
        expect(container.querySelector('i')).toHaveClass('las test');
     })
})
