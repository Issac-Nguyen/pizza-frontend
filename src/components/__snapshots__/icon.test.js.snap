// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should take a snapshot 1`] = `
<DocumentFragment>
  <i
    class="las test"
    data-testid="icon"
  />
</DocumentFragment>
`;
